### ISSUE TITLE
DeepFaceLabOpenCLSSE uses only CPU

### ISSUE THREAD

**Issue by henrikkelner** [iperov/DeepFaceLab#365]
## Expected behavior

I'm expecting the program to use the GPU when extracting faces and training.

## Actual behavior

It only uses the CPU and is very slow. It does this with every step, whether extracting faces or training.

## Steps to reproduce

I ran the train SAE batch file and used the default settings.
![cpu](https://user-images.githubusercontent.com/54379324/63476080-4264a800-c488-11e9-89d9-67533f4255d7.jpg)

## Other relevant information
I'm running Windows 7 Home Premium 64 bit. My GPU is AMD Radeon HD 6800 1024mb, the drivers as up to date as they can be and as far as I can tell my OpenCL dll version for my GPU is 2.0.4.0.

**Comment by lbfs** [comment]
AMD Radeon HD 6800 only supports OpenCL 1.1 from the documentation I can find on it, plaidML requires OpenCL 1.2. I may be wrong about this, but if this is the case, then you will not be able to use your GPU for training.

**Comment by vinniejames** [comment]
I'm having the same trouble, extract_faces is taking forever. 0% gpu usage, 9% CPU

I'm running Windows 10, Ryzen 3700X

GPU is AMD RX 480
Driver 19.8.1

In the Radeon settings app, I don't see OpenCL listed under the Software tab, but it is showing OpenCL API Version 2.0 under the Hardware tab

Any idea how I can speed things up?

**Comment by lbfs** [comment]
Seems like fairly normal speeds and completely unrelated to this issue.

**Comment by vinniejames** [comment]
Does extract not require much of the GPU resources? 1% usage feels low.

**Comment by s1rleblanc** [comment]
> ## Expected behavior
> I'm expecting the program to use the GPU when extracting faces and training.
> 
> ## Actual behavior
> It only uses the CPU and is very slow. It does this with every step, whether extracting faces or training.
> 
> ## Steps to reproduce
> I ran the train SAE batch file and used the default settings.
> ![cpu](https://user-images.githubusercontent.com/54379324/63476080-4264a800-c488-11e9-89d9-67533f4255d7.jpg)
> 
> ## Other relevant information
> I'm running Windows 7 Home Premium 64 bit. My GPU is AMD Radeon HD 6800 1024mb, the drivers as up to date as they can be and as far as I can tell my OpenCL dll version for my GPU is 2.0.4.0.

**i have the same issue, but im on macOS Mojave and my GPU is AMD Radeon Pro Vega 20
 4GB VRAM.

i tried something like this:

$ ./5_data_dst_extract_faces_S3FD_all_GPU.sh 
Performing 1st pass...
Running on CPU0.
Using TensorFlow backend.
Running on CPU1.
Using TensorFlow backend.
Running on CPU2.
Using TensorFlow backend.
Running on CPU3.
Using TensorFlow backend.
Running on CPU4.
Using TensorFlow backend.
Running on CPU5.
Using TensorFlow backend.
Running on CPU6.
Using TensorFlow backend.
Running on CPU7.
Using TensorFlow backend.
100%|###########################################################################################| 1538/1538 [10:34<00:00,  2.42it/s]
Performing 2nd pass...
Running on CPU0.
Using TensorFlow backend.


and make some test with plaidbench and there's work it...

$ plaidbench --batch-size 16 keras --train mobilenet
Running 1024 examples with mobilenet, batch size 16, on backend plaid
Loading CIFAR data
INFO:plaidml:Opening device "opencl_amd_radeon_pro_vega_20_compute_engine.0"
Compiling network...Epoch 1/1
INFO:plaidml:Analyzing Ops: 1381 of 2265 operations complete
16/16 [==============================] - 15s 939ms/step - loss: 9.6524 - acc: 0.0000e+00
 Warming up...Epoch 1/1
32/32 [==============================] - 1s 37ms/step - loss: 8.2260 - acc: 0.0312
 Running...
Epoch 1/1
1024/1024 [==============================] - 96s 94ms/step - loss: 2.4261 - acc: 0.4600
Example finished, elapsed: 15.207s (compile), 96.263s (execution)

-----------------------------------------------------------------------------------------
Network Name         Inference Latency         Time / FPS          
-----------------------------------------------------------------------------------------
mobilenet            94.01 ms                  0.00 ms / 709638.61 fps 

do i missing something at setting up my DFL ??.. thank you!!
**

**Comment by square690410** [comment]
Check below registry key path.
HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\OpenCL\Vendors

It should have value like below
"C:\\WINDOWS\\System32\\DriverStore\\FileRepository\\c0328279.inf_amd64_d32f2d9d2e8240b7\\B327689\\amdocl64.dll"=dword:00000000
"C:\\WINDOWS\\System32\\DriverStore\\FileRepository\\k131835.inf_amd64_459cc21b8c275ef0\\IntelOpenCL64.dll"=dword:00000000

if opencl doesn't find any value in this key path, it will use CPU

**Comment by square690410** [comment]
There are less guys using OpenCL. however , I met float convert problem when training above 8000 times. it's very terrible. sometime it can fix by changed to using CPU or Intel.
the other thing is convert process , it only works with CPU. I don't know why.

**Comment by jeremiahbyerley** [comment]
IDKW but the GPU activity is not showing on the Windows task manager. From the Windows start menu, open the AMD Radeon program. Click the performance tab, there you will see an accurate usage percentage.
